consumer/sm_context.go

```diff
@@ -24,16 +25,67 @@ import (
 	"github.com/omec-project/openapi/models"
 )
 
+func getServingSmfIndex(smfNum int) (servingSmfIndex int) {
+	servingSmfIndexStr := os.Getenv("SERVING_SMF_INDEX")
+	i, _ := strconv.Atoi(servingSmfIndexStr)
+	servingSmfIndexInt := i + 1
+	servingSmfIndex = servingSmfIndexInt % smfNum
+	os.Setenv("SERVING_SMF_INDEX", strconv.Itoa(servingSmfIndex))
+	return
+}
+
+func setAltSmfProfile(smCtxt *amf_context.SmContext) error {
+	ignoreSmfId := smCtxt.SmfID()
+	var altSmfInst []models.NfProfile
+	//iterate over nf instances to ignore failed NF
+	for _, inst := range smCtxt.SmfProfiles {
+		if inst.NfInstanceId != ignoreSmfId {
+			altSmfInst = append(altSmfInst, inst)
+		}
+	}
+
+	if len(altSmfInst) > 0 {
+		smCtxt.SmfProfiles = altSmfInst
+		nfProfile := altSmfInst[0]
```

Comment:
- IF we always take 0th NF profile then why we create all alternate SMF list ?

Reply:
In the code above, we maintain only good SMF instances in amf context and remove non-reachable SMF instances(ignored one). So, it is safe to take 1st(0th index) element from remaining alternate SMF instances